GooglePay.PaymentDataCryptography/GoogleKeyProvider.cs

```diff
@@ -94,16 +101,25 @@ private Task FetchKeysIfNeeded(bool force = false)
                 }
                 _googleKeysTask = FetchGoogleKeys();
             }
+            finally
+            {
+                _semaphoreSlim.Release();
+            }
             return WaitForKeys();
             async Task WaitForKeys()
             {
-                KeysDict keys = await _googleKeysTask.ConfigureAwait(false);
-                lock (_lock)
+                await _semaphoreSlim.WaitAsync();
```

Comment:
Same here: `await _semaphoreSlim.WaitAsync().ConfigureAwait(false);`

Reply:
Hey @fstanis - good call on adding `ConfigureAwait(false)`!

@mhentosh is no longer working with us so I'm not sure he'll follow up on this PR.  Instead, I opened a new PR #6 with your feedback.  Can you review that instead?  Thanks!